lib/player.js

```diff
@@ -2837,6 +2837,32 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     return this.config_;
   }
 
+  /**
+   * Returns the ratio of video length buffered compared to buffering Goal
+   * @return {number}
+   * @export
+   */
+  getBufferFullness() {
+    const bufferedLength = this.video_ ?
+        this.video_.buffered.length : 0;
+    const bufferedEnd =
+        bufferedLength ? this.video_.buffered.end(bufferedLength - 1) : 0;
+    const bufferingGoal = this.getConfiguration().streaming.bufferingGoal;
+    const lengthToBeBuffered = this.video_ ? Math.min(this.video_.currentTime +
+        bufferingGoal, this.seekRange().end) : 0;
+
+    if (this.video_) {
+      if (bufferedEnd >= lengthToBeBuffered) {
+        return 1;
+      } else if (bufferedEnd <= this.video_.currentTime) {
+        return 0;
+      } else if (bufferedEnd < lengthToBeBuffered) {
+        return ((bufferedEnd - this.video_.currentTime) / bufferingGoal);
```

Comment:
On further consideration, I think this is not quite right.  In the edge case where seekRange.end() limits buffering, this would not approach 1.0, because the buffered amount would never reach bufferingGoal.

I think it should probably be:

```js
    return (bufferedEnd - this.video_.currentTime) /
        (lengthToBeBuffered - this.video_.currentTime);
```

Does that make sense?

Reply:
Yes, it makes complete sense. I used this formula in my previous commit thinking about that edge case but when I rechecked later I couldn't recall the exception case and changed the formula. 
Thank you